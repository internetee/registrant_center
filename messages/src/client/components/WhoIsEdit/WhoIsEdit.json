[
  {
    "id": "whois.edit.check_some",
    "defaultMessage": "({count}/{countOf}) Osa info peidetud"
  },
  {
    "id": "whois.edit.check_all",
    "defaultMessage": "({count}/{countOf}) Kõik info nähtav"
  },
  {
    "id": "whois.edit.check_none",
    "defaultMessage": "({count}/{countOf}) Kõik info peidetud"
  },
  {
    "id": "domain.role",
    "defaultMessage": "Roll:"
  },
  {
    "id": "whois.edit.name",
    "defaultMessage": "Nimi avalik ({contactName})"
  },
  {
    "id": "whois.edit.email",
    "defaultMessage": "E-mail avalik ({contactEmail})"
  }
]