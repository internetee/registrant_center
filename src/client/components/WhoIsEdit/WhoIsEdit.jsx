/* eslint-disable camelcase */
import React from 'react';
import { FormattedHTMLMessage, FormattedMessage } from 'react-intl';
import { Form, Checkbox } from 'semantic-ui-react';
import staticMessages from '../../utils/staticMessages.json';

function WhoIsEdit({ contacts, isOpen, checkAll, lang, handleWhoIsChange }) {
  const contactsList = Object.values(contacts);
  const { totalCount, isCheckedAll, checkedCount } = contactsList.reduce((acc, { disclosed_attributes }) => ({
    checkedCount: acc.checkedCount + disclosed_attributes.size,
    isCheckedAll: acc.isCheckedAll + disclosed_attributes.size === acc.totalCount + 2,
    totalCount: acc.totalCount + 2,
  }), {
    checkedCount: 0,
    isCheckedAll: false,
    totalCount: 0,
  });
  
  const indeterminate = checkedCount > 0 && checkedCount < totalCount;
  
  const CheckAllLabel = () => {
    if (indeterminate) {
      return (
        <FormattedMessage
          id="whois.edit.check_some"
          defaultMessage="({ count }/{ countOf }) Osa info peidetud"
          tagName="label"
          values={{
            count: checkedCount,
            countOf: totalCount
          }}
        />
      );
    }
    if (isCheckedAll) {
      return (
        <FormattedMessage
          id="whois.edit.check_all"
          defaultMessage="({ count }/{ countOf }) Kõik info nähtav"
          tagName="label"
          values={{
            count: checkedCount,
            countOf: totalCount
          }}
        />
      );
    }
    return (
      <FormattedMessage
        id="whois.edit.check_none"
        defaultMessage="({ count }/{ countOf }) Kõik info peidetud"
        tagName="label"
        values={{
          count: checkedCount,
          countOf: totalCount
        }}
      />
    );
  };
  
  const onChange = (checked, ids, type) => {
    const changedContacts = ids.reduce((acc, id) => {
      const { disclosed_attributes } = contacts[id];
      const attributes = new Set(disclosed_attributes);
      type.forEach(attr => {
        if (checked) {
          attributes.add(attr);
        } else {
          attributes.delete(attr);
        }
      });
      return {
        ...acc,
        [id]: {
          disclosed_attributes: attributes
        }
      };
      
    }, {});
    handleWhoIsChange(changedContacts);
  };
  
  const Roles = ({ roles }) => {
    return [...roles].map((role, i) => {
      const { domain } = staticMessages[lang];
      if (i === roles.size - 1) {
        return domain[role];
      }
      if (i === roles.size - 2) {
        return `${domain[role]} & `;
      }
      return `${domain[role]}, `;
    });
  };
  
  return (
    <React.Fragment>
      { checkAll && (
        <Form.Field>
          <Checkbox
            className='large'
            checked={isCheckedAll}
            indeterminate={indeterminate}
            label={<CheckAllLabel />}
            onChange={(e, elem) => onChange(elem.checked,[...new Set(contactsList.map(item => item.id))],['name', 'email'])}
          />
        </Form.Field>
      )}
      <Form.Group grouped className={ isOpen ? 'adv-field-group u-visible' : 'adv-field-group' }>
        { contactsList.map(item => (
          <React.Fragment key={item.id}>
            <label htmlFor={item.id}>
              <FormattedHTMLMessage
                id="domain.role"
                defaultMessage="Roll: "
                tagName="strong"
              />
              <strong>
                <Roles roles={item.roles} />
              </strong>
            </label>
            <Form.Field>
              <Checkbox
                name='name'
                checked={item.disclosed_attributes.has('name')}
                value={item.name}
                label={<FormattedMessage
                  id="whois.edit.name"
                  defaultMessage="Nimi avalik ({ contactName })"
                  tagName="label"
                  values={{
                    contactName: item.initialName
                  }}
                />}
                onChange={(e, elem) => onChange(elem.checked,[item.id],['name'])}
              />
            </Form.Field>
            <Form.Field>
              <Checkbox
                name='email'
                checked={item.disclosed_attributes.has('email')}
                value={item.email}
                label={<FormattedMessage
                  id="whois.edit.email"
                  defaultMessage="E-mail avalik ({ contactEmail })"
                  tagName="label"
                  values={{
                    contactEmail: item.initialEmail
                  }}
                />}
                onChange={(e, elem) => onChange(elem.checked,[item.id],['email'])}
              />
            </Form.Field>
          </React.Fragment>
        ))}
      </Form.Group>
    </React.Fragment>
  );
}

export default WhoIsEdit;
