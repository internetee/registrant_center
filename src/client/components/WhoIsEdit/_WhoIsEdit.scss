.domains-grid {
  position: relative;
  z-index: 2;
  display: flex;
  align-items: stretch;
  flex-wrap: wrap;
  width: 1440px;
  padding: 0 30px;
  max-width: 100%;
  margin: -40px auto 0;
  &--item {
    flex: 0 1 50%;
    min-width: 50%;
    padding: 0 20px 40px;
    text-align: center;
    .content {
      padding: 40px 20px 20px;
      margin: auto 0;
    }
    a {
      display: flex;
      flex-direction: column;
      justify-content: center;
      min-height: 100%;
      color: #212224;
      background: #fff;
      box-shadow: 0 0 20px rgba(0,0,0,.1);
    }
    h2 {
      margin: 0 0 20px;
      font-size: 24px;
      line-height: 20px;
      font-weight: 900;
    }
    p {
      margin: 0;
    }
    .data {
      display: flex;
      justify-content: center;
      flex-wrap: wrap;
      padding: 10px;
      background: #EAEBED;
      strong {
        display: block;
      }
      p {
        flex: 1 0 50%;
        width: 50%;
        min-width: 200px;
        padding: 0 10px 10px;
      }
    }
  }
}
.domains-list {
  position: relative;
  z-index: 2;
  display: flex;
  align-items: stretch;
  width: 1440px;
  padding: 0 50px;
  max-width: 100%;
  margin: -40px auto 40px;
}