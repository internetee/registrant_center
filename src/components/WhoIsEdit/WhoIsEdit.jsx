/* eslint-disable camelcase */
import React from 'react';
import { FormattedMessage } from 'react-intl';
import { Form, Checkbox } from 'semantic-ui-react';
import Roles from '../Roles/Roles';

function WhoIsEdit({ contacts, isOpen, checkAll, onChange }) {
    const contactsList = Object.values(contacts);
    const { totalCount, isCheckedAll, checkedCount } = contactsList.reduce(
        (acc, { disclosed_attributes }) => ({
            checkedCount: acc.checkedCount + disclosed_attributes.size,
            isCheckedAll: acc.isCheckedAll + disclosed_attributes.size === acc.totalCount + 2,
            totalCount: acc.totalCount + 2,
        }),
        {
            checkedCount: 0,
            isCheckedAll: false,
            totalCount: 0,
        }
    );

    const indeterminate = checkedCount > 0 && checkedCount < totalCount;

    const handleChange = (checked, ids, type) => {
        const changedContacts = ids.reduce((acc, id) => {
            const { disclosed_attributes } = contacts[id];
            const attributes = new Set(disclosed_attributes);
            type.forEach((attr) => {
                if (checked) {
                    attributes.add(attr);
                } else {
                    attributes.delete(attr);
                }
            });
            return {
                ...acc,
                [id]: {
                    ...contacts[id],
                    disclosed_attributes: attributes,
                },
            };
        }, {});
        onChange(changedContacts);
    };

    const CheckAllLabel = () => {
        if (indeterminate) {
            return (
                <FormattedMessage
                    id="whois.edit.checkSome"
                    tagName="label"
                    values={{
                        count: checkedCount,
                        countOf: totalCount,
                    }}
                />
            );
        }
        if (isCheckedAll) {
            return (
                <FormattedMessage
                    id="whois.edit.checkAll"
                    tagName="label"
                    values={{
                        count: checkedCount,
                        countOf: totalCount,
                    }}
                />
            );
        }
        return (
            <FormattedMessage
                id="whois.edit.checkNone"
                tagName="label"
                values={{
                    count: checkedCount,
                    countOf: totalCount,
                }}
            />
        );
    };

    return (
        <>
            {checkAll && (
                <Form.Field>
                    <Checkbox
                        checked={isCheckedAll}
                        className="large"
                        indeterminate={indeterminate}
                        label={<CheckAllLabel />}
                        onChange={(e, elem) =>
                            handleChange(elem.checked, Object.keys(contacts), ['name', 'email'])
                        }
                    />
                </Form.Field>
            )}
            <Form.Group
                className={isOpen ? 'adv-field-group u-visible' : 'adv-field-group'}
                grouped
            >
                {contactsList.map((item) => (
                    <React.Fragment key={item.id}>
                        {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
                        <label htmlFor={item.id}>
                            <FormattedMessage id="domain.role" tagName="strong" />
                            <strong>
                                <Roles roles={item.roles} />
                            </strong>
                        </label>
                        <Form.Field>
                            <Checkbox
                                checked={item.disclosed_attributes.has('name')}
                                label={
                                    <FormattedMessage
                                        id="whois.edit.name"
                                        tagName="label"
                                        values={{
                                            contactName: item.initialName,
                                        }}
                                    />
                                }
                                name="name"
                                onChange={(e, elem) =>
                                    handleChange(elem.checked, [item.id], ['name'])
                                }
                                value={item.name}
                            />
                        </Form.Field>
                        <Form.Field>
                            <Checkbox
                                checked={item.disclosed_attributes.has('email')}
                                label={
                                    <FormattedMessage
                                        id="whois.edit.email"
                                        tagName="label"
                                        values={{
                                            contactEmail: item.initialEmail,
                                        }}
                                    />
                                }
                                name="email"
                                onChange={(e, elem) =>
                                    handleChange(elem.checked, [item.id], ['email'])
                                }
                                value={item.email}
                            />
                        </Form.Field>
                    </React.Fragment>
                ))}
            </Form.Group>
        </>
    );
}

export default WhoIsEdit;
