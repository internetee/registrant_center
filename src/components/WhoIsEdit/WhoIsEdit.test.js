/* eslint-disable react/prop-types */
import React from "react"
import { shallow } from "enzyme"
import WhoIsEdit from "./WhoIsEdit"
import mockUser from "../../__mocks__/user"
import mockDomains from "../../__mocks__/domains"
import mockContacts from "../../__mocks__/contacts"
import Helpers from "../../utils/helpers"

describe("components/WhoIsEdit", () => {
    it("should show content", () => {
        const contacts = Helpers.getDomainContacts(mockDomains[0], mockContacts)
        const test = Helpers.getDomains(mockDomains)
        const cmp = shallow(
            <WhoIsEdit contacts={contacts} domain={test[0]} user={mockUser} />
        )
        expect(cmp).toMatchSnapshot()
    })
})
