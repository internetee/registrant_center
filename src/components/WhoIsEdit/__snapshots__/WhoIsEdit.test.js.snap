// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/WhoIsEdit should show content 1`] = `ShallowWrapper {}`;
